### DISCUSSION TITLE
SW0 interrupt acts differently between USB and Battery power?

### DISCUSSION THREAD

**Discussion by PhatHub** [microchip-pic-avr-solutions/avr-iot-cellular-arduino-library#13] [Q&A] [answered]
Hello! I have an AVR-IoT Cellular Mini board, and I have two interrupts for the buttons:

1. `SW0` for interactivity (currently used to progress through the sketch)
2. `SW1` for "reset" (`jmp(0)`)

When connected to my computer via USB (or a USB power supply), I am able to use `SW0` 100% consistently.

When I disconnect the USB cable and run it on a LiPo battery, `SW0` is very inconsistent: 

- Sometimes it will work perfectly 
- Sometimes it will stop working, and then start working after a while. 
- Sometimes when it stops working it will stay in the malfunctioned "state" until I trigger `SW1` to reset the sketch.
- However, if it is in a "broken" state, it will work fine when I plug it into the USB power supply.

Meanwhile `SW1` (for resetting the sketch) works perfectly regardless of being powered by USB or a battery.

Was there a warning or anything I missed in the instructions that might cause issues with `SW0` under battery power?

Thanks!

**Comment by M65649** [comment, marked as answer]
Hello,

How do you configure SW0 and SW1? Could you post some example code? Do you have pullups on the buttons? I see that we’ve also forgot to add pull-ups for some of the examples in the library. We’ll fix that for the next release.

**Reply by PhatHub** [reply]
Hi!

Thanks for responding! I think you got my mistake... There was a difference in setting up the input pins:

```c
//set interrupts
pinConfigure(SW0, PIN_DIR_INPUT); //("SW0" is defined as PIN_PD2)
attachInterrupt(SW0, SW0_down, FALLING);

// Set PF6 as input (reset button)
pinConfigure(PIN_PF6, PIN_DIR_INPUT | PIN_PULLUP_ON);
attachInterrupt(PIN_PF6, resetInterrupt, FALLING);
```

So as shown above, I'll try or-ing `PIN_PULLUP_ON` and see how that affects my battery operation! Thanks!

Edit: So far, it's looking good. Thanks!
